JUPE7H/UI/Elements/Canvas.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace JUPE7H.UI.Elements;
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Structs;
+
+[SuppressMessage("Interoperability", "CA1416:Validate platform compatibility")]
+internal sealed class Canvas{
+    #region DLL import
+
+    [DllImport("Kernel32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
+    private static extern SafeFileHandle CreateFile(
+        string fileName,
+        [MarshalAs(UnmanagedType.U4)] uint fileAccess,
+        [MarshalAs(UnmanagedType.U4)] uint fileShare,
+        IntPtr securityAttributes,
+        [MarshalAs(UnmanagedType.U4)] FileMode creationDisposition,
+        [MarshalAs(UnmanagedType.U4)] int flags,
+        IntPtr template);
+
+    [DllImport("kernel32.dll", SetLastError = true)]
+    private static extern bool WriteConsoleOutputW(
+        SafeFileHandle hConsoleOutput,
+        CharInfo[] lpBuffer,
+        Point dwBufferSize,
+        Point dwBufferCoord,
+        ref Rectangle lpWriteRegion);
+
+    #endregion
+
+    private SafeFileHandle _handle{ get; }
+    private CharInfo[] _buffer{ get; set; }
+    private Point _size{ get; set; }
+    
+    public short Width => _size.X;
+    public short Height => _size.Y;
+
+    public Canvas(short width, short height){
+        _handle = CreateFile("CONOUT$", 0x40000000, 2, IntPtr.Zero, FileMode.Open, 0, IntPtr.Zero);
+        _size = new Point(width, height);
+        _buffer = Array.Empty<CharInfo>();
+        
+        Resize(width, height);
```

Comment:
Console.WindowWidth és Console.WindowHeight hívással kiváltható, ami multiplatform is.